tracer/src/Datadog.Tracer.Native/iast/method_info.cpp

```diff
@@ -109,7 +109,7 @@ namespace iast
 
     WSTRING MemberRefInfo::GetFullName(bool includeReturnType)
     {
-        if (_fullName.size() == 0)
+        if (_fullName.size() == 0 && _fullNameCounterLock.fetch_add(1, std::memory_order_acquire) == 0)
```

Comment:
If I've understood correctly, this isn't really a lock, as it's never released, but it will ensure the below code is executed once and only once?

Reply:
I first used a mutex, but Dani made a fair observation: since we would will need many locks, one per instance, it could make sense to use an atomic counter instead, to save resources. In this case, we guarantee that only the first concurrent thread will be able to enter the critical area. After that, the fullname will be filled and the other threads will not even try to increase the counter or enter the critical section.